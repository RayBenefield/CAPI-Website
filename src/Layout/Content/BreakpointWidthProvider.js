import React from "react";

/*
 * A simple HOC that provides facility for listening to container resizes.
 */
const BreakpointWidthProvider = (ComposedComponent) => class BreakpointWidthProvider extends React.Component {
    constructor(props) {
        super(props);
        this.defaultProps = {
            measureBeforeMount: false
        };
        this.state = {
            width: 1280
        };
        this.mounted = false;
    }

    componentDidMount() {
        this.mounted = true;

        window.addEventListener('resize', this.onWindowResize.bind(this));
        // Call to properly set the breakpoint and resize the elements.
        // Note that if you're doing a full-width element, this can get a little wonky if a scrollbar
        // appears because of the grid. In that case, fire your own resize event, or set `overflow: scroll` on your body.
        this.onWindowResize();
    }

    componentWillUnmount() {
        this.mounted = false;
        window.removeEventListener('resize', this.onWindowResize);
    }

    onWindowResize() {
        if (!this.mounted) return;

        const currentWidth = window.innerWidth;
        if (currentWidth > 1280 && this.state !== 1280) {
            this.setState({width: 1280});
        } else if (currentWidth > 960 && this.state !== 960) {
            this.setState({width: 960});
        } else if (currentWidth > 640 && this.state !== 640) {
            this.setState({width: 640});
        } else if (currentWidth > 320 && this.state !== 320) {
            this.setState({width: 320});
        }
    }

    render() {
        if (this.props.measureBeforeMount && !this.mounted) {
            return <div className={this.props.className} style={this.props.style} />;
        }

        return <ComposedComponent {...this.props} {...this.state} />;
    }
};

export default BreakpointWidthProvider;
