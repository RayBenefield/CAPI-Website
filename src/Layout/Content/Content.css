.App-intro {
    font-size: large;
}

.card {
    display: inline-block;
    margin: 10px;
    width: 380px;
}

.layout^ {
    margin: 10px;
}
