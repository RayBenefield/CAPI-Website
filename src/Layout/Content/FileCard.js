import _ from 'underscore';
import React from 'react';
import { Card, CardTitle, CardMedia } from 'material-ui/Card';

const MapCard = (props) => {
    const cardProps = _.omit(props, 'children', 'fileData');
    const { fileData: file } = props;
    return (
        <div {...cardProps}>
            <Card className="card">
                <CardMedia
                    overlay={<CardTitle title={file.name} style={{ 'font-size': '16px' }} />}
                >
                    <img src={file.image} alt='nature' />
                </CardMedia>
            </Card>
            {props.children}
        </div>
    )
};

export default MapCard;
