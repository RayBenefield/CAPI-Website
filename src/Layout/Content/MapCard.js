import _ from 'underscore';
import React from 'react';
import { Card, CardActions, CardMedia, CardTitle, CardText } from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import mapImg from './alpine.jpg';

const MapCard = (props) => {
    const cardProps = _.omit(props, 'children');
    return (
        <div {...cardProps}>
            <Card className="card">
                <CardMedia>
                    <img src={mapImg} alt='nature' />
                </CardMedia>
                <CardTitle title="Card title" subtitle="Card subtitle" />
                <CardText>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Donec mattis pretium massa. Aliquam erat volutpat. Nulla facilisi.
                    Donec vulputate interdum sollicitudin. Nunc lacinia auctor quam sed pellentesque.
                    Aliquam dui mauris, mattis quis lacus id, pellentesque lobortis odio.
                </CardText>
                <CardActions>
                    <FlatButton label="Action1" />
                    <FlatButton label="Action2" />
                </CardActions>
            </Card>
            {props.children}
        </div>
    )
};

export default MapCard;
