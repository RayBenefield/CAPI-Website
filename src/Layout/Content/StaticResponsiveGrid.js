import React from "react";
import { Responsive } from 'react-grid-layout';
import layouts from './layouts';

/*
 * A simple HOC that provides facility for listening to container resizes.
 */
const BreakpointWidthProvider = class BreakpointWidthProvider extends React.Component {
    constructor(props) {
        super(props);
        this.defaultProps = {
            measureBeforeMount: false
        };
        this.state = {
            width: 1280
        };
        this.mounted = false;
    }

    componentDidMount() {
        this.mounted = true;

        window.addEventListener('resize', this.onWindowResize.bind(this));
        // Call to properly set the breakpoint and resize the elements.
        // Note that if you're doing a full-width element, this can get a little wonky if a scrollbar
        // appears because of the grid. In that case, fire your own resize event, or set `overflow: scroll` on your body.
        this.onWindowResize();
    }

    componentWillUnmount() {
        this.mounted = false;
        window.removeEventListener('resize', this.onWindowResize);
    }

    onWindowResize() {
        if (!this.mounted) return;

        const currentWidth = window.innerWidth;
        if (currentWidth > 1600 && this.state !== 1600) {
            this.setState({width: 1601});
        } else if (currentWidth > 1280 && this.state !== 1280) {
            this.setState({width: 1281});
        } else if (currentWidth > 960 && this.state !== 960) {
            this.setState({width: 961});
        } else if (currentWidth > 640 && this.state !== 640) {
            this.setState({width: 641});
        } else if (currentWidth > 320 && this.state !== 320) {
            this.setState({width: 321});
        }
    }

    render() {
        if (this.props.measureBeforeMount && !this.mounted) {
            return <div className={this.props.className} style={this.props.style} />;
        }

        return (
            <div className="layout" style={{ width: `${this.state.width}px` }}>
                <Responsive
                    width={this.state.width}
                    className="layout"
                    autoSize={false}
                    layouts={layouts}
                    breakpoints={{lg: 1600, md: 1280, sm: 960, xs: 640, xxs: 320}}
                    cols={{lg: 5, md: 4, sm: 3, xs: 2, xxs: 1}}
                    rowHeight={169}
                >
                    {this.props.items}
                </Responsive>
            </div>
        );
    }
};

export default BreakpointWidthProvider;
