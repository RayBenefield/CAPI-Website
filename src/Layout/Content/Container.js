import { connect } from 'react-redux';
import Content from '.';

const mapStateToProps = (state) => {
    return {
        files: state.get('files').toArray(),
        item: state.getIn(['layout', 'item']),
    };
};

const ContentContainer = connect(
    mapStateToProps
)(Content);

export default ContentContainer;
