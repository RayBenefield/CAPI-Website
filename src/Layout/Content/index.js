import React, { Component } from 'react';
import './Content.css';

export default () => {
    return (
        <p className="App-intro">
            To get started, edit <code>src/App.js</code> and save to reload.
        </p>
    );
};
