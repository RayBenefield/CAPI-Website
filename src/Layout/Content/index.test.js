import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Content from '.';

it('renders without crashing', () => {
    const wrapper = shallow(<Content files={[]} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
