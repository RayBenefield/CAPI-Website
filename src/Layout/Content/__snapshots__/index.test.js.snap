exports[`test renders without crashing 1`] = `
<div
  className="content"
  style={
    Object {
      "height": "undefinedpx",
    }
  }>
  <div
    style={
      Object {
        "height": "20px",
      }
    } />
  <Connect(StaticResponsiveGrid)
    itemHeight={180.32573289599998}
    itemWidth={320}
    items={Array []}
    maxWidth={1600} />
</div>
`;
